python/rmm/_lib/device_buffer.pyx

```diff
@@ -350,45 +350,49 @@ cpdef void copy_ptr_to_host(uintptr_t db,
     """
 
     if hb is None:
-        with gil:
-            raise TypeError(
-                "Argument `hb` has incorrect type"
-                " (expected bytes-like, got NoneType)"
-            )
+        raise TypeError(
+            "Argument `hb` has incorrect type"
+            " (expected bytes-like, got NoneType)"
+        )
 
     cdef cudaError_t err
+    cdef cuda_stream_view c_stream
+
+    if stream is None:
+        c_stream = cuda_stream_view()
+    else:
+        c_stream = dereference(stream.c_obj.get())
+
+    with nogil:
+        err = cudaMemcpyAsync(<void*>&hb[0], <const void*>db, len(hb),
+                              cudaMemcpyDeviceToHost, <cudaStream_t>c_stream)
 
-    err = cudaMemcpyAsync(<void*>&hb[0], <const void*>db, len(hb),
-                          cudaMemcpyDeviceToHost, <cudaStream_t>stream)
     if err != cudaError.cudaSuccess:
-        with gil:
-            raise RuntimeError(f"Memcpy failed with error: {err}")
+        raise RuntimeError(f"Memcpy failed with error: {err}")
 
-    if stream == 0:
-        err = cudaStreamSynchronize(<cudaStream_t>stream)
-        if err != cudaError.cudaSuccess:
-            with gil:
-                raise RuntimeError(f"Stream sync failed with error: {err}")
+    if stream is None:
```

Comment:
We should also handle where the stream is explicitly the null stream.

Reply:
Explicitly the null stream? Or is any form of default stream? (per-thread, default, legacy default)

Reply:
Sorry, I meant the legacy default stream, but we could follow the semantics being discussed in https://github.com/numba/numba/pull/5162 where the only time a sync is done is when a stream isn't explicitly provided.

Reply:
If we want to do that I will have to go back to having None as an option...  Currently I'm in the process of making None not an option, so let me know.

Reply:
Starting a new thread since this is now outdated code